src/main/java/frc/robot/subsystems/pivot/Pivot.kt

```diff
@@ -1,181 +1,116 @@
+            Units.Degrees.of(90.0),
+            Units.Degrees.of(0.0),
+            Units.Degrees.of(90.0),
+            ControllerGains(
+                kP = 10.0, kI = 0.5, kD = 0.0,
+                kS = 0.0, kG = 0.43, kV = 1.34, kA = 0.02
+            )
+        )
 
-    /**
-     * Get the interpolated angle for a given position
-     * @param position The distance to get the angle for
-     * @return The angle for the given distance
-     */
-    fun getInterpolatedAngle(position: Double): Double {
-        return autoAimTree.get(position)
+        Constants.RobotConstants.Mode.REPLAY -> object : PivotIO {}
     }
 
-    /**
-     * Get the absolute position of the pivot
-     * @return The absolute position of the pivot
-     */
-    fun getAbsolutePosition(): Double {
-        return absoluteEncoder.distance
-    }
+    private val inputs = PivotIO.PivotInputs()
 
-    /**
-     * Get the relative position of the pivot
-     * @return The relative position of the pivot
-     */
-    fun getRelativePosition(): Double {
-        return relativeEncoder.position
-    }
+    val mechanism: Mechanism2d = Mechanism2d(3.0, 3.0)
+    val root = mechanism.getRoot("pivot", 2.0, 1.0)
+    val arm = root.append(MechanismLigament2d("arm", 0.5, 90.0))
 
-    /**
-     * Sync the relative encoder to the absolute encoder
-     */
-    fun syncRelative(){
-        relativeEncoder.setPosition(getAbsolutePosition())
-    }
+    private val currentSpikeTracker = LinearFilter.highPass(5.0, 0.02)
 
-    /**
-     * Get the homing sensor value
-     * @return The value of the homing sensor
-     */
-    fun getHomingSensor() = homingSensor.get()
-
-    /**
-     * Zero the relative encoder
-     */
-    fun zeroEncoder() {
-        relativeEncoder.setPosition(0.0)
-    }
+    fun manualControl(voltage: DoubleSupplier) = run {
+        io.setRawVoltage(
+            Units.Volts.of(voltage.asDouble),
+            false
+        )
+    }.withName("Manual Control")
 
-    /**
-     * Set the speed of the pivot
-     * @param speed The speed to set the pivot to
-     */
-    fun setRawSpeed(speed: Double){
-        pivotMotor.set(speed)
-    }
+    fun getHomeCommand() =
+        runOnce { io.setRawVoltage(Units.Volts.of(-3.0)) }
+            .andThen(WaitUntilCommand { inputs.isAtHardstop })
+            .andThen(runOnce { io.setKnownPosition(Units.Degrees.of(90.0)); io.stop() }).withName("Home Pivot")
 
-    /**
-     * Set a voltage amount to send to the pivot
-     * @param voltage The voltage to send to the pivot
-     */
-    fun setVoltage(voltage: Double) {
-        pivotMotor.setVoltage(voltage)
-    }
+    fun getSensorlessHomeCommand(threshold: Double = 5.5) =
+        runOnce { io.setRawVoltage(Units.Volts.of(-3.0)) }
+            .andThen(WaitUntilCommand { currentSpikeTracker.lastValue() > threshold })
+            .andThen(runOnce { io.setKnownPosition(Units.Degrees.of(90.0)); io.stop() }).withName("Sensorless Home Pivot")
```

Comment:
What happened to `Commands.sequence()`?

Reply:
I wrote this beforehand, neither is wrong, that's just what I find looks better, I didn't know about it yet